civiform/civiform
# Show modal with warning about universal questions when publishing a single program

### Description

This PR adds a modal that will be shown when a CiviForm admin attempts to publish a single program. If the admin has not used all universal questions in the program, the modal will show a warning communicating that. If the admin has used all universal questions in the program (or if there are no universal questions) the modal will show without the warning. This replaces current behavior where the admin is asked to confirm they want to publish via a browser dialogue popup. This functionality is behind the `UNIVERSAL_QUESTIONS` feature flag.


### Checklist

#### General


#### New Features


### Instructions for manual testing

1. Turn the `UNIVERSAL_QUESTIONS` feature flag on
2. Create a program
3. Create a question and do not set it as a universal question
4. Add the question to the program
5. Click the "Publish" button for the individual program
6. See the modal asking if you would like to publish 
7. Click "Cancel"
8. Create another question and set it to "universal"
9. Click the "Publish" button for the individual program
10. See the modal asking if you would like to publish the program with a warning that you have not used all universal questions
11. Click "Cancel"
12. Add the newly created universal question to the program
13. Click the "Publish" button for the individual program
14. See the modal asking if you would like to publish the program (warning should be gone now)
15. Click "Publish program"
16. See the active program

### Issue(s) this completes

Fixes #<issue_number>; Fixes #<issue_number>...
